### PR TITLE
MINOR:Fill missing parameter annotations for LogCleaner methods

### PR DESCRIPTION
**Pull request by hudeqi**
1. When reading the LogCleaner-related code, it is found that some methods do not update and improve the parameter annotations in time, and it is hoped that added annotations will be helpful to read the relevant logic.
2. LogSegment's append method does not return a value, so we should remove "@Return".

### CODE REVIEW

**Comment by hudeqi** [comment]
Hello, could you help to review this PR? @jlprat

**Comment by jlprat** [comment]
Thanks for the PR @hudeqi. I think it's always good to improve JavaDocs or ScalaDocs.

I was looking at the `LogCleaner.scala` file and I saw there are plenty of methods that are public which have only a very general documentation and they don't have any parameter annotation with documentation. For example `abortCleaning` in line 220 and some of the following methods.

For the sake of completion, would you be up to adding the missing annotations to the methods that are public? Extra mile for all the ones that are package-log-protected (`private[log]`)

**Comment by hudeqi** [comment]
> Thanks for the PR @hudeqi. I think it's always good to improve JavaDocs or ScalaDocs.
> 
> I was looking at the `LogCleaner.scala` file and I saw there are plenty of methods that are public which have only a very general documentation and they don't have any parameter annotation with documentation. For example `abortCleaning` in line 220 and some of the following methods.
> 
> For the sake of completion, would you be up to adding the missing annotations to the methods that are public? Extra mile for all the ones that are package-log-protected (`private[log]`)

Hi, thanks for the comments, I have added the comments of the methods you mentioned, please help to review, thank you! @jlprat

**Comment by hudeqi** [comment]
Hi, are you still following this pr? @jlprat

**Comment by jlprat** [comment]
Hi @hudeqi, it's on my list, yes.
I'll try to review this either today or tomorrow.

File: core/src/main/scala/kafka/log/LogCleaner.scala [hunk 1]
```diff
@@ -173,14 +174,25 @@ class LogCleaner(initialConfig: CleanerConfig,
     }
   }
 
+  /**
+   * Remove metrics when shutdown cleaner threads
```
Comment (jlprat) [inline core/src/main/scala/kafka/log/LogCleaner.scala]: I find this sentence a bit confusing, I think keeping it short is more than enough:
```suggestion
   * Remove metrics
```

File: core/src/main/scala/kafka/log/LogCleaner.scala [hunk 2]
```diff
@@ -173,14 +174,25 @@ class LogCleaner(initialConfig: CleanerConfig,
...
 
+  /**
+   * @return A set of configs that is reconfigurable in LogCleaner
+   */
   override def reconfigurableConfigs: Set[String] = {
     LogCleaner.ReconfigurableConfigs
   }
 
+  /**
+   * Mainly validate the new cleaner threads num is reasonable
```
Comment (jlprat) [inline core/src/main/scala/kafka/log/LogCleaner.scala]: I think we should avoid the use of `Mainly` in Javadoc (or Scaladoc). Having a "mainly" in there might imply that there are other things this method does, but are not being shared.

File: core/src/main/scala/kafka/log/LogCleaner.scala [hunk 3]
```diff
@@ -215,27 +230,37 @@ class LogCleaner(initialConfig: CleanerConfig,
   /**
    *  Abort the cleaning of a particular partition, if it's in progress. This call blocks until the cleaning of
    *  the partition is aborted.
+   *
+   *  @param topicPartition The topic and partition to be abort cleaning
```
Comment (jlprat) [inline core/src/main/scala/kafka/log/LogCleaner.scala]:
```suggestion
   *  @param topicPartition The topic and partition to abort cleaning
```

File: core/src/main/scala/kafka/log/LogCleaner.scala [hunk 4]
```diff
@@ -91,6 +91,7 @@ import scala.util.control.ControlThrowable
  * @param initialConfig Initial configuration parameters for the cleaner. Actual config may be dynamically updated.
  * @param logDirs The directories where offset checkpoints reside
  * @param logs The pool of logs
+ * @param logDirFailureChannel The channel used to add offline log dirs that may occur when cleaning the log
```
Comment (jlprat) [inline core/src/main/scala/kafka/log/LogCleaner.scala]: I think `occur` should be replaced with `be encountered`. But this seems to be a nitpick.

File: core/src/main/scala/kafka/log/LogCleaner.scala [hunk 5]
```diff
@@ -215,27 +230,37 @@ class LogCleaner(initialConfig: CleanerConfig,
...
+   *  @param topicPartition The topic and partition to be abort cleaning
    */
   def abortCleaning(topicPartition: TopicPartition): Unit = {
     cleanerManager.abortCleaning(topicPartition)
   }
 
   /**
    * Update checkpoint file to remove partitions if necessary.
+   *
+   * @param dataDir The file object to be updated
```
Comment (jlprat) [inline core/src/main/scala/kafka/log/LogCleaner.scala]: This is consistent with other JavaDocs in the code.

File: core/src/main/scala/kafka/log/LogCleaner.scala [hunk 6]
```diff
@@ -215,27 +230,37 @@ class LogCleaner(initialConfig: CleanerConfig,
...
+   * @param partitionToRemove The topicPartition to be removed, default none
    */
   def updateCheckpoints(dataDir: File, partitionToRemove: Option[TopicPartition] = None): Unit = {
     cleanerManager.updateCheckpoints(dataDir, partitionToRemove = partitionToRemove)
   }
 
   /**
-   * alter the checkpoint directory for the topicPartition, to remove the data in sourceLogDir, and add the data in destLogDir
+   * Alter the checkpoint directory for the topicPartition, to remove the data in sourceLogDir, and add the data in destLogDir
```
Comment (jlprat) [inline core/src/main/scala/kafka/log/LogCleaner.scala]: ScalaDoc allows to use backticks when you refer to variable names (or a paramers) in the doc.
```suggestion
   * Alter the checkpoint directory for the `topicPartition`, to remove the data in `sourceLogDir`, and add the data in `destLogDir`
```

File: core/src/main/scala/kafka/log/LogCleaner.scala [hunk 7]
```diff
@@ -803,6 +864,14 @@ private[log] class Cleaner(val id: Int,
     growBuffers(maxSize)
   }
 
+  /**
+   * Judge a batch should be discard by cleaned transaction state
```
Comment (jlprat) [inline core/src/main/scala/kafka/log/LogCleaner.scala]: I find "Judge" a bit odd in this situation.
```suggestion
   * Determine if a batch should be discard by cleaned transaction state
```

File: core/src/main/scala/kafka/log/LogCleaner.scala [hunk 8]
```diff
@@ -803,6 +864,14 @@ private[log] class Cleaner(val id: Int,
...
+   *
+   * @param batch The batch of records to judge
```
Comment (jlprat) [inline core/src/main/scala/kafka/log/LogCleaner.scala]:
```suggestion
   * @param batch The batch of records to cehck
```

File: core/src/main/scala/kafka/log/LogCleaner.scala [hunk 9]
```diff
@@ -811,6 +880,18 @@ private[log] class Cleaner(val id: Int,
       transactionMetadata.onBatchRead(batch)
   }
 
+  /**
+   * Judge a record should be retained
```
Comment (jlprat) [inline core/src/main/scala/kafka/log/LogCleaner.scala]: Same as before
```suggestion
   * Determine if a record should be retained
```

File: core/src/main/scala/kafka/log/LogCleaner.scala [hunk 10]
```diff
@@ -811,6 +880,18 @@ private[log] class Cleaner(val id: Int,
...
+   *
+   * @param map The offset map(key=>offset) to use for cleaning segments
+   * @param retainDeletesForLegacyRecords Should tombstones (lower than version 2) and markers be retained while cleaning this segment
+   * @param batch The batch of records that the record belongs to
+   * @param record The record to judge
```
Comment (jlprat) [inline core/src/main/scala/kafka/log/LogCleaner.scala]:
```suggestion
   * @param record The record to check
```

File: core/src/main/scala/kafka/log/LogSegment.scala [hunk 11]
```diff
@@ -140,7 +140,6 @@ class LogSegment private[log] (val log: FileRecords,
    * @param largestTimestamp The largest timestamp in the message set.
    * @param shallowOffsetOfMaxTimestamp The offset of the message that has the largest timestamp in the messages to append.
    * @param records The log entries to append.
-   * @return the physical position in the file of the appended records
```
Comment (jlprat) [inline core/src/main/scala/kafka/log/LogSegment.scala:143]: Good catch

**Review by jlprat — COMMENTED** [review]
Thanks @hudeqi it's looking good. I have some comments on some sentences used.

File: core/src/main/scala/kafka/log/LogCleaner.scala (hunk 8 above)
Comment (hudeqi) [reply core/src/main/scala/kafka/log/LogCleaner.scala]: mean 'check'?

Comment (jlprat) [reply core/src/main/scala/kafka/log/LogCleaner.scala]: Yes, sorry, I did a typo.

File: core/src/main/scala/kafka/log/LogCleaner.scala (hunk 7 above)
Comment (jlprat) [reply core/src/main/scala/kafka/log/LogCleaner.scala]: You could also use "check" instead of "determine"

**Review by jlprat — APPROVED** [review]
LGTM! Thanks @hudeqi

**Pull request merged**